index.js

```diff
@@ -111,6 +116,51 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+        var pluginRequiresBaseDir = true;
+
+        // do not look for properties on string plugins
+        if (Array.isArray(plugin)) {
+          // when an array, the second argument is an options hash
+          pluginRequiresBaseDir = false;
+          hash.plugins.push(stringify(plugin[1]));
+        } else if (pluginType === 'function' || pluginType === 'object') {
+          pluginAugmentsCacheKey = typeof plugin.cacheKey === 'function';
+          pluginProvidesBaseDir = typeof plugin.baseDir === 'function';
+        } else if (pluginType === 'string') {
+          pluginRequiresBaseDir = false;
+          hash.plugins.push(plugin);
+        }
```

Comment:
What happens when plugin is not and array, function object or string rather undefined symbol number boolean or some funky host object.